tmt/steps/execute/__init__.py

```diff
@@ -1416,3 +1418,42 @@ def results_for_tests(
             for test_origin in tests
             if test_origin.test.serial_number not in referenced_serial_numbers
         ]
+
+    def _assert_required_tests_executed(self) -> None:
+        """
+        Assert all required tests were executed.
+        """
+
+        results = set()
+        expected_results = {}
+        for result, test_origin in self.results_for_tests(self.plan.discover.required_tests):
+            if test_origin is None:
+                continue
```

Comment:
Definitely worth mentioning, as a warning at least: it's a result for which we do not have a test. Weird, isn't it?